onnxconverter_common/auto_mixed_precision_model_path.py

```diff
@@ -0,0 +1,276 @@
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+###########################################################################
+
+"""
+This tool converts a model to mixed precision (float32->float16) while excluding nodes as needed to maintain
+a certain accuracy. After the conversion, the model will be saved on the disk under given path. 
+A model with a size > 2G should leverage this.
+
+Example usage:
+
+    from onnxconverter_common import auto_mixed_precision_model_path
+    import numpy as np
+    import time
+
+    source_model_path = "/home/user/onnx/mymodel/model/8_fp32/graph.onnx"
+    target_model_path = '/home/user/onnx/mymodel/output/fp16.onnx'  # better to specify an %output% folder
+    location = "fp16_tensor.data"
+
+
+    # Could also use rtol/atol attributes directly instead of this
+    def validate(res1, res2):
+        for r1, r2 in zip(res1, res2):
+            if not np.allclose(r1, r2, rtol=0.01, atol=0.001):
+                return False
+        return True
+
+    auto_convert_mixed_precision_model_path(
+        source_model_path, test_data,
+        target_model_path, location=None,
+        customized_validate_func=None, rtol=None, atol=None,
+        keep_io_types=True, providers=None)
+
+You don't need to call onnx.save_model() in the customer code.
+The target fp16 model and its external data will be saved as %target_model_path and %location.
+Please note that keep_io_types=True, so the test_data can be used for both fp32 and fp16 models during the convertion.
+
+"""
+
+import copy
+import numpy as np
+import onnxruntime as ort
+import onnx
+import os
+import uuid
+from onnxconverter_common import float16
+from onnx import shape_inference
+from .auto_mixed_precision import SegmentList
+
+
+def auto_convert_mixed_precision_model_path(source_model_path, input_feed,
+                                            target_model_path, location=None,
+                                            customized_validate_func=None, rtol=None, atol=None,
+                                            keep_io_types=True, providers=None, verbose=False):
+    """
+    Automatically converts a model to mixed precision, excluding the minimum number of nodes required to
```

Comment:
Suggest:

Automatically converts a model to mixed precision, excluding the minimum number of nodes required to     ensure valudate_fn returns True and/or results are equal according to rtol/atol and saves the converted model on the disk. This function requires the source model's path as an input (source_model_path), so it still works well when the model's size > 2G.